core/src/miner/mem_pool.rs

```diff
@@ -216,33 +216,8 @@ impl MemPool {
         }
     }
 
-    pub fn top_transactions(&self, gas_limit: usize, size_limit: usize, range: Range<u64>) -> PendingTransactions {
-        let mut current_gas: usize = 0;
-        let mut current_size: usize = 0;
-        let unordered_transactions: Vec<_> =
-            self.transaction_pool.pool.values().filter(|tx| range.contains(&tx.inserted_timestamp)).collect();
-        let ordered_transactions = self.tx_filter.fetch_transactions_for_block(&unordered_transactions);
-        let chosen_transactions: Vec<_> = ordered_transactions
-            .iter()
-            .take_while(|tx_with_gas| {
-                let size = tx_with_gas.size();
-                let gas = tx_with_gas.gas;
-                current_size += size;
-                current_gas += gas;
-                current_gas < gas_limit && current_size < size_limit
-            })
-            .collect();
-
-        let last_timestamp =
-            chosen_transactions.iter().map(|tx_with_gas| tx_with_gas.tx_with_metadata.inserted_timestamp).max();
-
-        PendingTransactions {
-            transactions: chosen_transactions
-                .iter()
-                .map(|tx_with_gas| tx_with_gas.tx_with_metadata.tx.clone())
-                .collect(),
-            last_timestamp,
-        }
+    pub fn all_pending_transactions_with_metadata(&self) -> Vec<TransactionWithMetadata> {
```

Comment:
I'm worried about this costly clone.
The best solution would be creating a slice from `transaction_pool.pool`, but unfortunately, it was impossible.
I'm still looking for a nice solution.

Reply:
This is why I put **do_not_merge** label.
If you think this is fine, I can just merge it.
Otherwise, I'll continue looking for a solution.

Reply:
I think it's okay to merge this commit with leaving the `XXX` comment if fixing it takes too much time.